libs/cri-response-service/src/main/java/uk/gov/di/ipv/core/library/service/CriResponseService.java

```diff
@@ -62,15 +66,36 @@ public void persistCriResponse(
         dataStore.create(criResponseItem, ConfigurationVariable.CRI_RESPONSE_TTL);
     }
 
-    public CriResponseItem getFaceToFaceRequest(String userId) {
-        return dataStore.getItem(userId, "f2f");
-    }
-
     public void deleteCriResponseItem(String userId, Cri credentialIssuer) {
         dataStore.delete(userId, credentialIssuer.getId());
     }
 
     public void updateCriResponseItem(CriResponseItem responseItem) {
         dataStore.update(responseItem);
     }
+
+    public AsyncCriStatus getAsyncResponseStatus(
+            String userId,
+            Function<Cri, Boolean> hasVc,
+            Function<Cri, Long> getVcIat,
+            boolean isPendingEvcsIdentity) {
+        final var criResponseItem =
+                dataStore.getItems(userId).stream()
+                        .filter(item -> ASYNC_CRIS.contains(Cri.fromId(item.getCredentialIssuer())))
+                        .findFirst()
+                        .orElse(null);
+        if (Objects.isNull(criResponseItem)) {
+            return new AsyncCriStatus(null, null, null, false, false);
+        }
+
+        var cri = Cri.fromId(criResponseItem.getCredentialIssuer());
+
+        var awaitingVc = !hasVc.apply(cri);
+        var isCriPrecededByOtherRequiredVcs = F2F.equals(cri);
+        var isComplete =
+                hasVc.apply(cri) && (!isCriPrecededByOtherRequiredVcs || isPendingEvcsIdentity);
```

Comment:
Have removed the configService.enabled(EVCS_...) check, meaning this PR must wait for PYIC-6082